### PR TITLE
release 0.10.0 & improve README & add `update_test_file`

### PR DESCRIPTION
**Pull request by xxchan**
When I'm updating `risinglight`'s test files, I found `update_record_with_output` is still not handy enough, so I added `update_test_file`, which is basically the same code in CLI. 😇

### CODE REVIEW

**Review by skyzh — APPROVED** [review]

**Review by wangrunji0408 — APPROVED** [review]

**Comment by xxchan** [comment]
cc @alamb @xudong963 in case you need it.

This is how I update risinglight's tests:

image

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,20 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.10.0] - 2022-12-15
+
+- Improve the ability to unparse and update the test files. Mainly add `update_record_with_output` and `update_test_file` to the library.
```
Comment (alamb) [inline CHANGELOG.md:12]: 👍

File: sqllogictest/src/runner.rs [hunk 2]
```diff
@@ -1030,6 +1030,144 @@ pub fn update_record_with_output(
...
+/// Updates a test file with the output produced by a Database. It is an utility function wrapping
+/// [`update_test_file_with_runner`].
+///
+/// Specifically, it will create `"{filename}.temp"` to buffer the updated records and then override the
+/// original file with it.
+///
+/// Some other notes:
+/// - empty lines at the end of the file are cleaned.
+/// - `halt` and `include` are correctly handled.
+pub async fn update_test_file<D: AsyncDB>(
```
Comment (alamb) [inline sqllogictest/src/runner.rs:1042]: Thank you for this @xxchan  -- I will attempt to integrate this upstream into DataFusion's sqllogictest runner hopefully this weekend

👍

**Review by alamb — COMMENTED** [review]
Thank you @xxchan !

**Pull request merged**